### PR TITLE
Add Json Extract

### PR DESCRIPTION
**Pull request by joel**
Related to http://jmeter.apache.org/usermanual/component_reference.html#Regular_Expression_Extractor

### CODE REVIEW

**Comment by s12chung** [comment]
hmm..... i think it'd be more elegant to have the idea of `json_var`.

basically, in json:
```json
{ uuid: 'json_var(uuid_regexp)' } // make it explicit that it's ruby
```

then:
```
RailsJmeter.json_vars = {
   uuid_regexp: /some_regex/
}
```

that way, you can be more DRY and write regex in ruby. plus you get standard variables. and then... see my other comments.

File: lib/rails_jmeter/ruby_jmeter/rails.rb [hunk 1]
```diff
@@ -37,10 +41,15 @@ def assert_response(item, json_path: "", array_limit: nil)
           end
         else
           item = item.nil? ? "null" : item
-          assert json: json_path, value: item, name: "#{json_path} == #{item.to_json}"
+
+          if item.to_s.match(/^REGEXP: /)
+            extract json: json_path, name: item.gsub('REGEXP: ','')
```
Comment (s12chung) [inline lib/rails_jmeter/ruby_jmeter/rails.rb]: i'd change `item.gsub('REGEXP: ','')` to something like "#{json_path} ~= #{the_regexp}"

File: lib/rails_jmeter/ruby_jmeter/rails.rb [hunk 2]
```diff
@@ -37,10 +41,15 @@ def assert_response(item, json_path: "", array_limit: nil)
           end
         else
           item = item.nil? ? "null" : item
-          assert json: json_path, value: item, name: "#{json_path} == #{item.to_json}"
+
```
Comment (s12chung) [inline lib/rails_jmeter/ruby_jmeter/rails.rb]:
```ruby
item = get_json_var(item)

if item.class == Regex
   extract json: json_path, name: "#{json_path} ~= #{item}"
else.....
```

Comment (s12chung) [inline lib/rails_jmeter/ruby_jmeter/rails.rb]: wait.... this is an `extract` (saving a variable from a regex), not an `assert` (comparing values), basically...... do you want to do:

https://github.com/flood-io/ruby-jmeter#response-assertion

```ruby
assert matches: regex.to_s
```

instead?

Comment (s12chung) [inline lib/rails_jmeter/ruby_jmeter/rails.rb]: i'd add a note in the docs about it being `Perl5-style regular expressions`

File: lib/rails_jmeter/ruby_jmeter/rails.rb (hunk 1 above)
Comment (s12chung) [inline lib/rails_jmeter/ruby_jmeter/rails.rb]: ah, i'd change `~=` to be `match` because `~=` does not mean exact match

**Comment by thomas-riboulet-finalcad** [comment]
can we get this merged ? @yld @joel

**Comment by thomas-riboulet-finalcad** [comment]
please let's pick one CI and stick with it; mixing two or more is a pain to handle with configurations when you have a pipeline for packages

**Comment by joel** [comment]
That was a try, but it turned out JMeter provides a way to play with regex, no need the other mechanism.

**Pull request closed without merging**